ansible/user-management/tasks/linux-add-user.yml

```diff
@@ -0,0 +1,21 @@
+---
+- name: Checking if the user already exists in Linux
+  command: getent passwd {{ users.linux.username }}
+  register: user_check
+  ignore_errors: true
+  when: "ansible_os_family == 'RedHat' or ansible_os_family == 'Debian'"
+
+- debug:
+    var: user_check
+
+- name: Creating new user in Linux
+  user:
+    name: "{{ users.linux.username }}"
+    state: present
+    groups: "{{ users.linux.groups }}"
+    password: "{{ users.linux.password }}"
```

Comment:
pls dont use password based login for Linux users. Key based is preferred. You can keep an option to use password if provided else to set it to null.

Reply:
Removing the password section and adding an option to copy the ssh pub key to the authorized key section.